src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,59 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_NUMBER = 100;
+    private static Type type;
+
+    public FigureSupplier(Type type) {
+        this.type = type;
+    }
+
+    public static Figure getRandomFigure() {
+        Random random = new Random();
+        Random random3 = new Random();
+        Random random2 = new Random();
+        int value1 = random.nextInt(MAX_NUMBER) + 1;
+        int value2 = random.nextInt(MAX_NUMBER) + 1;
+        int value3 = random.nextInt(MAX_NUMBER) + 1;
+        int index = random.nextInt(Type.values().length);
+        type = Type.values()[index];
+        ColorSupplier colorSupplier = new ColorSupplier();

```

Comment:
same, let's make Random and ColorSupplier as class level variables, not to create new instances every time we call getRandomFigure() method.